src/chapters/control-flow-and-collections/list.rst

```diff
@@ -0,0 +1,343 @@
+      sum += grades[i];
+   }
+
+.. index:: ! List.Count
+
+Here, we introduce the syntax ``students.Count`` which utilizes the ``Count``
+property of ``List``. This property holds the integer representing the
+number of items in the List. This is similar to string's ``.length()`` method.
+
+In this ``for`` loop, we use a *loop index* to define the starting point,
+ending point, and increment for iteration. It may be helpful for you to
+consider this kind of construction as something like,  ``for integer i in the
+range of the number of items in students...``. The first statement inside the
+parenthesis declares and initializes a loop index variable ``i``. The second
+statement is a Boolean expression that is our exit condition. In other words,
+we will keep looping as long as this expression evaluates to ``true``. The
+third statement is used to increment the value of the loop index variable at
+the end of iteration through the loop.
+
+Again, the syntax ``i++`` is C# shorthand for ``i = i + 1``. C# also
+supports the shorthand ``i--`` to decrement the value of ``i``.
+We can also write ``i += 2`` as shorthand for ``i = i + 2``.
+
+In the final lines of the program, we compute the average grade for all
+students:
+
+.. sourcecode:: csharp
+   :lineno-start: 43
+
+   double avg = sum / students.Count;
+   Console.WriteLine("Average grade: " + avg);
+
+List Methods
+-----------------
+
+Let’s gather up a few of the ``List`` methods that we’ve encountered so
+far, along with a few new ones. While these will be the most common methods and
+properties that you use with this class, they by no means represent a complete
+List. Refer to the `official documentation on the List
+class <https://docs.microsoft.com/en-us/dotnet/api/system.collections.List?view=netframework-4.8>`__
+for such a List, and for more details.
+
+To demonstrate the use of these methods, we'll create a new ``List``
+called ``planets``.
+
+.. sourcecode:: csharp
+
+   List<string> planets = new List<string>();
+
+Ok, we've got an empty List. We need to use the class's ``.Add()`` method
+to populate this collection with items.
+
+.. admonition:: Note
+
+   There are other means to declare and initialize an List in fewer lines.
+   These require knowledge of other collections types, so we'll stick with ``.Add()``
+   for the time being.
+
+Using ``.Add()`` to populate ``planets``:
+
+.. sourcecode:: C#
+   :linenos:
+
+   planets.Add("Mercury");
+   planets.Add("Venus");
+   planets.Add("Earth");
+   planets.Add("Mars");
+   planets.Add("Jupiter");
+   planets.Add("Saturn");
+   planets.Add("Uranus");
+   planets.Add("Neptune");
+
+Thus, the first item in this table:
+
+.. _list-methods:
+
+.. list-table:: List methods in C#
+   :header-rows: 1
+
+   * - C# Syntax
+     - Description
+     - Example
+   * - ``Add()``
+     - Adds an item to the List
+     - ``planets.Add("Pluto")`` adds ``Pluto`` to ``planets``
+   * - ``Count``
```

Comment:
i know you already did this above, but it could help to pull this out as not a method but a property

Reply:
Done!